modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCAuthzCodeIdTokenValidationWhenSessionExistsTestCase.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.wso2.identity.integration.test.oidc;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.NameValuePair;
+import org.apache.http.message.BasicNameValuePair;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.wso2.identity.integration.test.utils.DataExtractUtil;
+import org.wso2.identity.integration.test.utils.OAuth2Constant;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class OIDCAuthzCodeIdTokenValidationWhenSessionExistsTestCase extends OIDCAuthzCodeIdTokenValidationTestCase {
```

Comment:
Extending an existing test will run its test again in this class also. Get the common methods to an abstract class and extend both these classes from that abstract class

Reply:
The main objective in here is to run the supper class once and again run the same methods in the same session. So creating an abstract class won't achieve that objective.